packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -166,9 +214,13 @@ const handleFocus = (event: FocusEvent) => {
 };
 
 const handleBlur = (event: FocusEvent) => {
-  handleSelect(autoSelectOption?.option);
-  setMenuState(CLOSED);
-  onBlur?.(event);
+  // blur can still be triggered if the input is diabled or the menu is closed,
+  // but we shouldn't select anything if so
+  if (!disabled && menuState !== CLOSED) {
+    handleSelect(autoSelectOption?.option);
+    setMenuState(CLOSED);
+    onBlur?.(event);
```

Comment:
Consider moving this out of the `if`. I would be surprised if I didn't receive a blur event, even if the menu is disabled

Reply:
61af29e

Good point